### PR TITLE
[Merged by Bors] - chore: more `refine'` to `refine`

### PR DESCRIPTION
**Pull request by adomani**
A PR analogous to #13166.  The reason some of these `refine'` are now caught is that my previous script missed all instances.  I am not sure why it is still missing other `refine'`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by adomani** [comment]
!bench

File: Mathlib/Algebra/Order/Module/OrderedSMul.lean
Comment (mattrobball) [inline Mathlib/Algebra/Order/Module/OrderedSMul.lean:124]: Is this one removable?

File: Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean
Comment (mattrobball) [inline Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean:203]: This one also?

File: Mathlib/NumberTheory/ClassNumber/Finite.lean
Comment (mattrobball) [inline Mathlib/NumberTheory/ClassNumber/Finite.lean:301]: Backwards?

File: Mathlib/Algebra/Homology/HomotopyCategory/ShiftSequence.lean
Comment (rosborn) [inline Mathlib/Algebra/Homology/HomotopyCategory/ShiftSequence.lean:122]: I feel like this should be `apply` (or perhaps we should discuss deprecating `apply` in favor of `refine`).

File: Mathlib/Algebra/Polynomial/Basic.lean
Comment (rosborn) [inline Mathlib/Algebra/Polynomial/Basic.lean:366]: Also `apply`, although I'm going to stop mentioning it now.

**Comment by adomani** [comment]
Matt, thanks for the review!  Somewhere, I must have done something wrong with a git commit.  I'm about to get into a plane, but will fix/answer later tonight!

**Review by rosborn — APPROVED** [review]
Other than Matt's comments this looks correct.

File: Mathlib/Algebra/Order/Module/OrderedSMul.lean
Comment (grunweg) [reply Mathlib/Algebra/Order/Module/OrderedSMul.lean:124]: I checked: it isn't.

File: Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean
Comment (grunweg) [reply Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean:203]: This one, however, is:
```suggestion
  refine ⟨Equiv.ofPreimageEquiv ?_, fun a => Equiv.ofPreimageEquiv_map _ _⟩

File: Mathlib/NumberTheory/ClassNumber/Finite.lean
Comment (grunweg) [reply Mathlib/NumberTheory/ClassNumber/Finite.lean:301]:
```suggestion
    refine ⟨⟨J, ?_⟩, ?_, ?_⟩

Comment (grunweg) [reply Mathlib/NumberTheory/ClassNumber/Finite.lean:301]: The refine' below is *not* removable; could this be the source of the error (that *all* `refine'`s in that proof were reverted)?

Comment (mattrobball) [reply Mathlib/NumberTheory/ClassNumber/Finite.lean:301]: This is not a good sign. Metavariable shenanigans might be happening here

**Comment by grunweg** [comment]
I took the liberty to push a commit replacing a handful of `refine` (in files modified) with apply. Feel free to revert, of you disagree with it.

**Review by grunweg — COMMENTED** [review]
I've read through it all and confirmed that all this PR does is replacing refine' by refine.
(and sometimes `refine stuff ?_` by `apply stuff`, where `stuff` could be several arguments).

(And `Mathlib/Algebra/BigOperators/Group/Multiset.lean`, and comment-only change.)
Benchmarking was neutral (though doesn't include the last batch); if any performance surprises come up, the last commit is easy to revert.

maintainer merge

File: Mathlib/NumberTheory/ClassNumber/Finite.lean
Comment (grunweg) [reply Mathlib/NumberTheory/ClassNumber/Finite.lean]: To follow up on this: this change is now removed from the diff: the diff in that file looks entirely reasonable.

It's very possible meta-variable shenanigans happen here - but my impression is that the current change should not perturb them.

**Comment by jcommelin** [comment]
bors d+

**Comment by riccardobrasca** [comment]
Since bors if free after the current batch.

bors r+ p=37

**Comment by riccardobrasca** [comment]
Mmm, the conflict is solved, isn't it?

**Comment by adomani** [comment]
Should be, let me try again to merge master

**Comment by adomani** [comment]
Yes, I just pulled and merged origin/master with no conflicts.  Should I push again?

**Comment by riccardobrasca** [comment]
I don't see any conflict, so I removed the label by hand.

**Comment by adomani** [comment]
Oh, this is going to be very quick, though, since the cache that it build was already essentially equal to the one on master.  In fact, it is already in the testing phase.

**Comment by adomani** [comment]
Once this is merged, I'll merge master on the other one and push again.

**Pull request closed without merging**
